test/Microsoft.AspNet.Diagnostics.Tests/ElmPageTest.cs

```diff
@@ -0,0 +1,210 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNet.FileProviders;
+using Xunit;
+
+namespace Microsoft.AspNet.Diagnostics
+{
+    public class ElmPageTest
+    {
+        [Theory]
+        [InlineData("TestFiles/SourceFile.txt")]
+        [InlineData(@"TestFiles\SourceFile.txt")]
+        public void DisplaysSourceCodeLines_ForAbsolutePaths(string path)
+        {
+            // Arrange
+            var rootPath = Directory.GetCurrentDirectory();
+            var absoluteFilePath = Path.Combine(rootPath, path);
+            // PhysicalFileProvider handles only relative paths but we fall back to work with absolute paths too
+            var provider = new PhysicalFileProvider(rootPath);
+
+            // Act
+            var stackFrame = ErrorPageMiddleware.GetStackFrame(
+                "func1",
+                absoluteFilePath,
+                lineNumber: 10,
+                fileProvider: provider,
+                sourceCodeLineCount: 6);
+
+            // Assert
+            // Lines 4-16 (inclusive) is the code block
+            Assert.Equal(4, stackFrame.PreContextLine);
+            Assert.Equal<string>(GetCodeLines(4, 9), stackFrame.PreContextCode);
```

Comment:
nah not C# compiler `PreContextCode` is an `IEnumerable<string>` XUnit is the one who has to be smart here.